### PR TITLE
Adding javascript ast visitor extension

### PR DESCRIPTION
**Pull request by JustinRudat**

### CODE REVIEW

File: extensions/js/src/main/java/io/doov/js/ast/ScriptEngineFactory.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+    public static ScriptEngine create() {
+        ScriptEngineManager sem = new ScriptEngineManager();            // creation of an engine manager
+        ScriptEngine engine = sem.getEngineByName("nashorn");        // engine creation based on nashorn
+        try{
+
+            InputStream stream = ScriptEngineFactory.class.getResourceAsStream(MOMENT_JS_SRC);
+            InputStreamReader momentJS = new InputStreamReader(stream);
+            engine.eval(momentJS);              // evaluating moment.js
+        }catch(ScriptException se){
```
Comment (dubreuia) [inline extensions/js/src/main/java/io/doov/js/ast/ScriptEngineFactory.java]: formatting

File: sample/validation/src/test/java/io/doov/sample/validation/js/engine/EngineTest.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        try {
+            engine.eval(mockValue); // evaluating the mock values for testing purpose
+        } catch (ScriptException e) {
+            e.printStackTrace();
+        }
+        ByteArrayOutputStream ops = new ByteArrayOutputStream();
+
+        final int[] index = new int[1];                                 // index as a tab, usage in lambda expression
+        final int[] counter = new int[1];
+        index[0]=0;
```
Comment (dubreuia) [inline sample/validation/src/test/java/io/doov/sample/validation/js/engine/EngineTest.java]: formatting

File: sample/validation/src/test/java/io/doov/sample/validation/js/engine/EngineTest.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        counter[0]=0;
+        REGISTRY_DEFAULT.stream().forEach(rule -> {
+            ops.reset();
+            try {
+                index[0]++;
+                System.out.println("--------------------------------\n");
+                new AstJavascriptVisitor(ops, BUNDLE, Locale.ENGLISH).browse(rule.metadata(), 0);
+                String request = new String(ops.toByteArray(), Charset.forName("UTF-8"));
+                try {
+                    if( index[0]!=14 ) {                                // excluding some rules for now (lambda expression)
```
Comment (dubreuia) [inline sample/validation/src/test/java/io/doov/sample/validation/js/engine/EngineTest.java]: fomatting

File: sample/validation/src/test/java/io/doov/sample/validation/ast/RulesVisitorTest.java [hunk 4]
```diff
@@ -14,15 +14,13 @@
```
Comment (ozangunalp) [inline sample/validation/src/test/java/io/doov/sample/validation/ast/RulesVisitorTest.java]: No modification is needed to this file

File: extensions/js/build.gradle [hunk 5]
```diff
@@ -0,0 +1,5 @@
+dependencies {
+    compile project(':doov-core')
+    compile project(':doov-assertions')
+    compile group: 'org.webjars', name: 'momentjs', version: '2.10.3'
+}
```
Comment (ozangunalp) [inline extensions/js/build.gradle]: No new line EOF

**Pull request merged**